File: spring-boot-api/src/main/java/dev/codewithsam/springbootapi/payments/Payment.java
package dev.codewithsam.springbootapi.payments;

import java.time.Instant;

/**
 * The domain object. Kept separate from the request on purpose: keeping
 * transport input away from the thing you store is a habit that transfers to
 * NestJS unchanged.
 */
public record Payment(
    String id,
    Long amountInMinorUnits,
    String currency,
    String idempotencyKey,
    Instant createdAt) {}

File: nestjs-api/src/payments/payment.ts
// The domain object. Deliberately separate from the request: keeping transport
// input away from the thing you store is a habit that transfers from Spring
// unchanged.
export interface Payment {
  id: string;
  amountInMinorUnits: number;
  currency: string;
  idempotencyKey: string;
  createdAt: string;
}
